### PR TITLE
[pyright] examples/with-airflow

### PR DESCRIPTION
**Pull request by smackesey**
### Summary & Motivation

Remove strict jinja2 pin and fix some type errors in `examples/with-airflow`. Strict jinja 2 pin caused difficulty installing in same env with other packages.

### How I Tested These Changes

BK

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11908** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11908" target="_blank">Graphite</a> 
    * **PR #11909** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11909" target="_blank">Graphite</a> 
      * **PR #11910** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11910" target="_blank">Graphite</a> 
        * **PR #11911** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11911" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11911?utm_source=stack-comment).

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11908** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11908" target="_blank">Graphite</a> 
    * **PR #11909** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11909" target="_blank">Graphite</a> 
      * **PR #11910** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11910" target="_blank">Graphite</a> 
        * **PR #11911** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11911" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11911?utm_source=stack-comment).

**Review by sryza — COMMENTED** [review]
I think it's worth getting @Ramshackle-Jamathon 's take on the pin, because he added it

File: examples/with_airflow/setup.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
         "dagster_airflow",
         "apache-airflow==2.3.0",
         # pin jinja2 to version compatible with dagit and airflow
-        "jinja2==3.0.3",
+        "jinja2",
```
Comment (Ramshackle-Jamathon) [inline examples/with_airflow/setup.py:11]: you should be able to drop this entirely

**Review by Ramshackle-Jamathon — APPROVED** [review]

**Pull request merged**